merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/ClaimSimulationAction.java

```diff
@@ -0,0 +1,35 @@
+package gov.nasa.jpl.aerie.merlin.server.remotes.postgres;
+
+import org.intellij.lang.annotations.Language;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/*package local*/ public class ClaimSimulationAction implements AutoCloseable {
+  private static final @Language("SQL") String sql = """
+    update simulation_dataset
+      set
+        state = 'incomplete',
+	      reason = ''
+      where (dataset_id = ? and state = 'pending');
+  """;
+
+  private final PreparedStatement statement;
+
+  public ClaimSimulationAction(final Connection connection) throws SQLException {
+    this.statement = connection.prepareStatement(sql);
+  }
+
+  public void apply(final long datasetId) throws SQLException {
+    this.statement.setLong(1, datasetId);
+
+    final var count = this.statement.executeUpdate();
+    if (count != 1) throw new SQLException("Failed to claim simulation.");
```

Comment:
🔵 Idle thought - since this is expected, rather than exceptional behavior, I wonder if it would make sense to return an `Optional<Long>` or something to indicate that the claim action failed. I have no strong reason to go either way.

If we do decide that an exception is the preferred mechanism to communicate a failed claim, my next thought is whether we want a specific exception for this scenario 🤔